### DISCUSSION TITLE
No packages loaded for Verdaccio 5.1.0 hosted on IIS over HTTPS

### DISCUSSION THREAD

**Discussion by pavelbednar** [verdaccio/verdaccio#3477] [Q&A]
**Describe the bug**
After upgrade from verdaccio 4.x.x to 5.1.0 webpage hosted on IIS is not showing packages. It work when approaching over HTTP. But if user goes to HTTPS version, verdaccio shows empty repository with information how to add packages.

When verdaccio is accessed via `https://npm.company.org` it tries to get api requests over HTTP (instead of HTTPS as in previous version). I.E. `http://npm.company.org/-/verdaccio/packages`. The request is stopped beause CSP no mixed content is allowed and no data is returned. 
In same configuration in 4.x.x it worked beause it sent api request via HTTPS. Configuration is same, only `npm install verdaccio@5.1.0` was called to upgrade.

**To Reproduce**
[1] Configure app as in documentation (https://verdaccio.org/docs/en/iss-server
[2] In IIS Manager create New site for npm repo with http binding to port 80 and https binding to port 443
[3] start the application

Update: tested also with only one binding over HTTPS to 443 (no http support). Website still does not run. First request to https://localhost succeeds - it returns html for verdaccio page which initiates requests for `runtime.xxxxx.js`, `vendors.xxxxxx.js` and `main.xxxxxx.js` but all of them over HTTP and none of them is resolved by IIS since HTTP binding does not exists.

**Expected behavior**
When oppening page over HTTPS protocol it should do HTTPS request to verdaccio API

**Screenshots**

**config.yaml**
```yaml
storage: d:\Web\Verdaccio\storage 
plugins: ./plugins
url_prefix: /


auth:
  htpasswd:
    file: ./htpasswd
    max_users: -1

uplinks:
  npmjs:
    url: https://registry.npmjs.org/
    cache: false
    cache_path: ./cache
```
**start.js**
```js
process.argv.push('-l', 'unix:' + process.env.PORT, '-c', './config.yaml'); 
require('./node_modules/verdaccio/build/lib/cli.js');
```

**Environment information**
Tested on Windows Server with latest IIS, also in latest updated Win10 with local IIS.
Tested in latest Chrome. Firefox has same CSP problem but it allows request to continue, thus Firefox shows packages even on HTTPS but it has same as chrome in console.

I tried to adjust IIS CSP header, upgrade insecure request and allow connect-src and script-src to HTTP but it didnt help.

**Comment by juanpicado** [comment]
Maybe https://verdaccio.org/blog/2021/04/14/verdaccio-5-migration-guide#after-upgrade-i-dont-see-packages-on-the-ui

**Comment by pavelbednar** [comment]
> Maybe https://verdaccio.org/blog/2021/04/14/verdaccio-5-migration-guide#after-upgrade-i-dont-see-packages-on-the-ui

Thx, but not the case (i just tried). In production environment we have storage on shared folder with full UNC path. Problem is really in diffrerent HTTP protocol. It should access api via HTTPS when browsing HTTPS page.

**Comment by juanpicado** [comment]
For me is really hard to test IIS Manager since I don't have it, maybe someone else is able to help here.

**Comment by bednarp** [comment]
> For me is really hard to test IIS Manager since I don't have it, maybe someone else is able to help here.

Its quite simple if you have windows 10 pro or higher. IIS can be added as a feature.
Start searching "Turn windows features on and off" and tick "Internet information services". (https://www.lansweeper.com/knowledgebase/how-to-install-iis/)
